numpy/core/src/multiarray/item_selection.c

```diff
@@ -1749,6 +1750,263 @@ PyArray_LexSort(PyObject *sort_keys, int axis)
     return NULL;
 }
 
+/*NUMPY_API
+ *Keysort an array providing indices that will sort a collection of arrays
+ *lexicographically.  The first key is sorted on first, followed by the second key
+ *-- requires that arg"merge"sort is available for each sort_key
+ *
+ *Returns None on success, returns NULL on failure
+ */
+NPY_NO_EXPORT PyObject *
+PyArray_KeySort(PyArrayObject *self, PyObject *sort_keys, int axis)
+{
+    PyArrayObject **mps;
+    PyArrayIterObject **its;
+    PyArrayIterObject *sit = NULL;
+    npy_intp n, N, size, i, j;
+    npy_intp astride, self_stride;
+    int nd;
+    int needcopy = 0;
+    int elsize;
+    int self_elsize;
+    int maxelsize;
+    int any_key_is_object = 0;
+    PyArray_ArgSortFunc *argsort;
+    NPY_BEGIN_THREADS_DEF;
+
+    if (!PySequence_Check(sort_keys)
+           || ((n = PySequence_Size(sort_keys)) <0)) {
+        PyErr_SetString(PyExc_TypeError,
+                "Error when trying to get length of sort keys.");
+        return NULL;
+    }
+    if (n == 0) {
+        //No keys provided, do nothing.
+        Py_RETURN_NONE;
+    }
+    mps = (PyArrayObject **) PyArray_calloc(n, sizeof(PyArrayObject *));
```

Comment:
```suggestion
    mps = PyArray_calloc(n, sizeof(*mps));
```
Casting the result of malloc in C is frowned upon.